docs/integrations/spark.rst

```diff
@@ -0,0 +1,98 @@
+Lance ❤️ Spark
+--------------------
+
+Lance can be used as a third party datasource of `<https://spark.apache.org/docs/latest/sql-data-sources.html>`_
+
+.. warning::
+    This feature is experimental and the APIs may change in the future.
+
+Build from source code
+~~~~~~~~~~~~~~~~~~
+.. code-block:: shell
+    git clone https://github.com/lancedb/lance.git
+    cd lance/java
+    mvn clean package -DskipTests -Drust.release.build=true
+
+After building the code, the spark related jars are under path :meth:`lance/java/spark/target/jars/`
+.. code-block:: shell
+	arrow-c-data-15.0.0.jar
+	arrow-dataset-15.0.0.jar
+	jar-jni-1.1.1.jar
+	lance-core-0.25.0-SNAPSHOT.jar
+	lance-spark-0.25.0-SNAPSHOT.jar
+
+
+
+Download the pre-build jars
+~~~~~~~~~~~~~~~~~~
+If you did not want to get jars from source, you can download these five jars from maven repo.
+.. code-block:: shell
+	wget https://repo1.maven.org/maven2/com/lancedb/lance-core/0.23.0/lance-core-0.23.0.jar
+	wget https://repo1.maven.org/maven2/com/lancedb/lance-spark/0.23.0/lance-spark-0.23.0.jar
+	wget https://repo1.maven.org/maven2/org/questdb/jar-jni/1.1.1/jar-jni-1.1.1.jar
+	wget https://repo1.maven.org/maven2/org/apache/arrow/arrow-c-data/12.0.1/arrow-c-data-12.0.1.jar
+	wget https://repo1.maven.org/maven2/org/apache/arrow/arrow-dataset/12.0.1/arrow-dataset-12.0.1.jar
+
+
+Configurations for Lance Spark Connector
+~~~~~~~~~~~~~~~~~~
+There are some configurations you have to set in `spark-defaults.conf` to enable lance datasource.
+.. code-block:: text
+	spark.sql.catalog.lance com.lancedb.lance.spark.LanceCatalog
+
+This config define the `LanceCatalog` and then the spark will treat lance as a datasource.
+
+If dealing with lance dataset stored in object store, these configurations should be set:
+.. code-block:: text
+	spark.sql.catalog.lance.access_key_id {your object store ak}
+	spark.sql.catalog.lance.secret_access_key {your object store sk}
+	spark.sql.catalog.lance.aws_region {your object store region}
```

Comment:
This is optional.

Reply:
OK, I write an `optional` for this configuration